### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

**Comment by sindresorhus** [comment]
You need to mark this PR as "Require Node.js 8" and take it out of `.travis.yml`.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
+declare namespace dargs {
+	interface Options {
+		/**
+		Keys or regex of keys to exclude. Takes precedence over `includes`.
+		*/
+		excludes?: ReadonlyArray<string | RegExp>;
```
Comment (sindresorhus) [inline index.d.ts]: Since it will be a major release anyway, you can change to `readonly`.

Comment (BendingBender) [reply index.d.ts]: Didn't we want to wrap complex types in `Array`/`ReadonlyArray`?

**Comment by BendingBender** [comment]
> You need to mark this PR as "Require Node.js 8" and take it out of `.travis.yml`.

I thought that we leave Node.js 6 for packages that are already at this level. Do you want to move to Node.js 8 even for those packages?

**Comment by sindresorhus** [comment]
> I thought that we leave Node.js 6 for packages that are already at this level. Do you want to move to Node.js 8 even for those packages?

Yes, but you already did. You cannot use object spread in Node.js 6: https://github.com/sindresorhus/dargs/pull/38/files#diff-168726dbe96b3ce427e7fedce31bb0bcR14 Or `Object.entries()`: https://github.com/sindresorhus/dargs/pull/38/files#diff-168726dbe96b3ce427e7fedce31bb0bcR42 Probably best to just revert those two changes then.

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts]: So `readonly Array<string | RegExp>` doesn't work?

**Pull request merged**